pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -510,38 +496,29 @@ class HassTypeHintChecker(BaseChecker):  # type: ignore[misc]
 
     def __init__(self, linter: PyLinter | None = None) -> None:
         super().__init__(linter)
-        self.current_package: str | None = None
-        self.module: str | None = None
+        self._function_matchers: list[TypeHintMatch] = []
 
     def visit_module(self, node: astroid.Module) -> None:
         """Called when a Module node is visited."""
-        self.module = node.name
-        if node.package:
-            self.current_package = node.name
-        else:
-            # Strip name of the current module
-            self.current_package = node.name[: node.name.rfind(".")]
+        self._function_matchers = []
+        for key, pattern in _MODULE_FILTERS.items():
+            if pattern.match(node.name):
+                if matches := _FUNCTION_MATCH.get(key):
+                    self._function_matchers.extend(matches)
```

Comment:
This will only ever be done once, for now. It should be safe to assign the reference instead of extending the list. Especially since it's read-only after this point.
```suggestion
                    self._function_matchers = matches
```
```

Reply:
done